src/openrct2/scripting/bindings/world/ScTileElement.cpp

```diff
@@ -63,6 +65,7 @@ namespace OpenRCT2::Scripting
 
     void ScTileElement::type_set(std::string value)
     {
+        RemoveBannerEntry();
```

Comment:
Yes, and the same goes for all the other affected properties.
Do you think this is a problem? Performance-wise certainly not and it will not happen a lot anyway; only if the plugin dev is careless.

Reply:
So here is where I stand on this, I think we can probably go with those changes for now but we should deprecate the usage of this API and remove it at some point, this hinders us from separating the data from the tile elements in the future which we have planned for quite some time now and keeping it like this would constantly add more and more checks to ensure associated data is removed.

Reply:
Sounds reasonable, but which part of the API exactly are you refering to? Changing `TileElement.type`? I think the only reason why this has to exist is because `Tile.insertElement` always insert a surface element.

Reply:
I'm talking about type, users should replace the element to promote it to another type which leaves only to places to deal with the data.

Reply:
I fully agree with that.

Reply:
Great, I don't expect you to do this in this PR but we should address this as soon as possible and give plugin authors the time to fix their code after deprecating it, perhaps we can provide a utility function for swapping tile elements which underneath will be just delete/insert into the same position on the tile stack.